tests/lib/uuid/src/main.c

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright (c) 2024, SECO Mind Srl
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr/ztest.h>
+#include <zephyr/sys/uuid.h>
+
+ZTEST(uuid, test_uuid_v4)
+{
+	uuid_t uuid_1 = {0};
+	uuid_t uuid_2 = {0};
+
+	uuid_generate_v4(uuid_1);
+	/* Check version and variant fields */
+	zassert_equal(uuid_1[6U] >> 4U, 4U, "Generated UUID v4 contains an incorrect 'ver' field");
+	zassert_equal(uuid_1[8U] >> 6U, 2U, "Generated UUID v4 contains an incorrect 'var' field");
+	/* Check another generated UUID is different */
+	uuid_generate_v4(uuid_2);
+	zassert(memcmp(uuid_1, uuid_2, UUID_SIZE) != 0, "The two generated UUID v4 are equal.");
+}
+
+ZTEST(uuid, test_uuid_v5)
+{
+	uuid_t namespace;
+	int result;
+	uuid_t uuid;
+	char uuid_str[UUID_STR_LEN + 1];
+
+	result = uuid_from_string("6ba7b810-9dad-11d1-80b4-00c04fd430c8", namespace);
+	zassert_true(result == 0, "uuid_from_string returned an error");
+	result = uuid_generate_v5(namespace, "www.example.com", 15, uuid);
+	zassert_true(result == 0, "uuid_generate_v5 returned an error");
+	result = uuid_to_string(uuid, uuid_str, UUID_STR_LEN + 1);
+	zassert_true(result == 0, "uuid_from_string returned an error");
+	zassert_true(strcmp("2ed6657d-e927-568b-95e1-2665a8aea6a2", uuid_str) == 0,
+		     "uuid_str != 2ed6657d-e927-568b-95e1-2665a8aea6a2");
```

Comment:
```suggestion
	zassert_str_equal("2ed6657d-e927-568b-95e1-2665a8aea6a2", uuid_str));
```

We actually have functions to compare strings (keeping it as is, is also fine)